Index: django_restapi/model_resource.py
===================================================================
--- django_restapi/model_resource.py	(revision 75)
+++ django_restapi/model_resource.py	(working copy)
@@ -96,7 +96,8 @@
         if not self.authentication.is_authenticated(request):
             response = self.responder.error(request, 401)
             challenge_headers = self.authentication.challenge_headers()
-            response._headers.update(challenge_headers)
+            for k,v in challenge_headers.items():
+                response[k] = v
             return response
         
         # Remove queryset cache
